lib/Runtime/Language/JavascriptOperators.cpp

```diff
@@ -9639,6 +9639,91 @@ namespace Js
         return superCtor;
     }
 
+    Var JavascriptOperators::OP_ImportCall(__in JavascriptFunction *function, __in Var specifier, __in ScriptContext* scriptContext)
+    {
+        ModuleRecordBase *moduleRecordBase = nullptr;
+        SourceTextModuleRecord *moduleRecord = nullptr;
+
+        FunctionBody* parentFuncBody = function->GetFunctionBody();
+        JavascriptString *specifierString = nullptr;
+
+        try
+        {
+            specifierString = JavascriptConversion::ToString(specifier, scriptContext);
+        }
+        catch (const JavascriptException &err)
+        {
+            Var errorObject = err.GetAndClear()->GetThrownObject(scriptContext);
+            if (errorObject != nullptr && Js::JavascriptError::Is(errorObject))
+            {
+                return SourceTextModuleRecord::ResolveOrRejectDynamicImportPromise(false, errorObject, scriptContext);
+            }
```

Comment:
Changed to assert + internal error.

---
In reply to: [115331615](https://github.com/Microsoft/ChakraCore/pull/2913#discussion_r115331615) [](ancestors = 115331615)